Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -459,8 +462,35 @@ def _addToLog(self, logName, message):
     def evaluateCommand(self, cmd):
         rc = super().evaluateCommand(cmd)
         steps_to_add = self.follow_up_steps()
-        if rc in (SUCCESS, WARNINGS) and self.getProperty('user_provided_git_hash'):
-            steps_to_add += [ArchiveMinifiedBuiltProduct(), UploadMinifiedBuiltProduct(), TransferToS3(terminate_build=True)]
+
+        triggers = self.getProperty('triggers', None)
+        if triggers:
+            # Build archive
+            steps_to_add += [ArchiveBuiltProduct()]
+            if CURRENT_HOSTNAME in BUILD_WEBKIT_HOSTNAMES + TESTING_ENVIRONMENT_HOSTNAMES:
+                steps_to_add.extend([
+                    GenerateS3URL(f"{self.getProperty('fullPlatform')}-{self.getProperty('architecture')}-{self.getProperty('configuration')}"),
+                    UploadFileToS3(f"WebKitBuild/{self.getProperty('configuration')}.zip", links={self.name: 'Archive'}),
+                ])
+            else:
+                # S3 might not be configured on local instances, achieve similar functionality without S3.
+                steps_to_add.extend([UploadBuiltProduct(), TransferToS3(terminate_build=True)])
+
+            # Minified build archive
+            full_platform = self.getProperty('fullPlatform')
+            if (full_platform.startswith('mac') or full_platform.startswith('ios-simulator') or full_platform.startswith('tvos-simulator')
+               or full_platform.startswith('watchos-simulator')) or (rc in (SUCCESS, WARNINGS) and self.getProperty('user_provided_git_hash')):
```

Comment:
There is still a small behaviour change when `user_provided_git_hash` is provided. So, the use-case is that someone want to build an archive for a specific hash and manually starts a build, in that case we want to ensure that we upload the archive, irrespective of whether that builder has triggers configured or not.

So, to achieve the similar logic, you can simplify this if condition to just have the `full_platform.startswith`. Then, for the `if triggers` part, add something like this:
```
elif (rc in (SUCCESS, WARNINGS) and self.getProperty('user_provided_git_hash')):
    steps_to_add.extend([
                        GenerateS3URL(f"{self.getProperty('fullPlatform')}-{self.getProperty('architecture')}-{self.getProperty('configuration')}", minified=True),
                        UploadFileToS3(f"WebKitBuild/minified-{self.getProperty('configuration')}.zip", links={self.name: 'Minified Archive'}),
                    ])
```

The reasoning being, that if there are triggers configured, we anyways upload an archive, but if not, than in case of `user_provided_git_hash ` we upload the minified archive (uploading either minified or full archive is fine for `user_provided_git_hash`). @ryanhaddad please let me know if I am missing something for that use-case.

Reply:
Do we want to have a full archive _and_ a minified archive in the case of there being triggers and `user_provided_git_hash`? I think that's how the existing logic works, rather than either minified or full archive being fine. If it doesn't matter whether the user gets full or minified then I'll implement it your way.

Reply:
This also minimizes code duplication:

```python
elif (triggers and full_platform.startswith(('mac', 'ios-simulator', 'tvos-simulator', 'watchos-simulator'))) 
     or (rc in (SUCCESS, WARNINGS) and self.getProperty('user_provided_git_hash')):
            # Minified build archive
            if full_platform.startswith(('mac', 'ios-simulator', 'tvos-simulator', 'watchos-simulator')):
                steps_to_add += [ArchiveMinifiedBuiltProduct()]
                if CURRENT_HOSTNAME in BUILD_WEBKIT_HOSTNAMES + TESTING_ENVIRONMENT_HOSTNAMES:
                    steps_to_add.extend([
                        GenerateS3URL(f"{full_platform}-{architecture}-{configuration}", minified=True),
                        UploadFileToS3(f"WebKitBuild/minified-{configuration}.zip", links={self.name: 'Minified Archive'}),
                    ])
                else:
                    steps_to_add.extend([UploadMinifiedBuiltProduct()])
```

Reply:
This looks good.